### PR TITLE
Tic tac toe

### LINKED ISSUE #97: tic tac toe
## Is your feature request related to a problem?

nein
## Describe the feature you'd like

tic with the tac and the toe
## Describe alternatives you've considered

no alternatives
## Additional context

nein

### PR DESCRIPTION
**Pull request by IllusionMan1212**
closes #97

### CODE REVIEW

**Comment by IllusionMan1212** [comment]
somebody please review :imdone:

File: src/commands/games/TicTacToe.ts [hunk 1]
```diff
@@ -0,0 +1,533 @@
...
+    /// Game Logic Functions ////
+    /////////////////////////////
+
+    getValidMoves(game: Game): Move[] {
+        const moves: Move[] = [];
+
+        game.board.forEach((row, rowNum) => {
+            return row.map((player, idx) => {
+                if (!player) {
+                    moves.push(<Move>{ row: rowNum, idx });
```
Comment (Sv443) [inline src/commands/games/TicTacToe.ts:240]: woah what is this funky shit

**Review by Sv443 — APPROVED** [review]
> toe

File: src/commands/games/TicTacToe.ts [hunk 2]
```diff
@@ -0,0 +1,533 @@
...
+        };
+
+        if (humanOpponent) { // le dumb human
+            await int.channel?.send({
+                content: `${int.user} has challenged ${humanOpponent} to a game of tic tac toe.`
+            })
+                .then((message) => {
+                    const filter = (_: MessageReaction, user: User) => user.id === humanOpponent.id;
+
+                    message.react("✅");
```
[the remark this answers is not in the document]
Comment (IllusionMan1212) [reply src/commands/games/TicTacToe.ts:185]: Who cares about the order lol

File: src/commands/games/TicTacToe.ts [hunk 3]
```diff
@@ -0,0 +1,533 @@
...
+                                this.drawBoard(game);
+                                await this.renderBoard(game, user.id);
+                                await this.sendBoard(int, game, user.id);
+                            } else {
+                                return await this.editReply(int, embedify(`${humanOpponent.username} has rejected your challenge`));
+                            }
+                        });
+                    });
+                })
+                .catch(console.error);
```
[the remark this answers is not in the document]
Comment (IllusionMan1212) [reply src/commands/games/TicTacToe.ts:204]: Im too lazy, we can just ctrl+f the entire project for console logs and errors in the future

File: src/commands/games/TicTacToe.ts (hunk 1 above)
Comment (IllusionMan1212) [reply src/commands/games/TicTacToe.ts:240]: Casting without the as keyword

File: src/commands/games/TicTacToe.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Sv443) [reply src/commands/games/TicTacToe.ts:185]: js

File: src/commands/games/TicTacToe.ts (hunk 1 above)
Comment (Sv443) [reply src/commands/games/TicTacToe.ts:240]: ts magic

File: src/commands/games/TicTacToe.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sv443) [reply src/commands/games/TicTacToe.ts:204]: creringe

**Pull request merged**